internal/pkg/cli/svc_package.go

```diff
@@ -123,14 +124,71 @@ func newPackageSvcOpts(vars packageSvcVars) (*packageSvcOpts, error) {
 		var serializer stackSerializer
 		switch t := mft.(type) {
 		case *manifest.LoadBalancedWebService:
-			var opts []stack.LoadBalancedWebServiceOption
-			if app.RequiresDNSDelegation() {
-				if err := validateLBSvcAliasAndAppVersion(aws.StringValue(t.Name), t.Alias, app, env.Name, appVersionGetter); err != nil {
+			if err := validateLBWSRuntime(app, env.Name, t, appVersionGetter); err != nil {
+				return nil, err
+			}
+			var options []stack.LoadBalancedWebServiceOption
+			if !t.NLBConfig.IsEmpty() {
+				envDescriber, err := describe.NewEnvDescriber(describe.NewEnvDescriberConfig{
+					App:         app.Name,
+					Env:         env.Name,
+					ConfigStore: store,
+					DeployStore: deployStore,
+				})
+				if err != nil {
+					return nil, fmt.Errorf("create describer for environment %s in application %s: %w", env.Name, app.Name, err)
+				}
+				envSession, err := p.FromRole(env.ManagerRoleARN, env.Region)
+				if err != nil {
+					return nil, fmt.Errorf("assuming environment manager role: %w", err)
+				}
+				subnetLister := ec2.New(envSession)
+				cidrBlocks, err := publicCIDRBlocks(envDescriber, subnetLister, env.Name)
+				if err != nil {
 					return nil, err
 				}
-				opts = append(opts, stack.WithHTTPS())
+				options = append(options, stack.WithNLB(cidrBlocks))
+				if app.RequiresDNSDelegation() {
+					resources, err := opts.appCFN.GetAppResourcesByRegion(app, env.Region)
+					if err != nil {
+						return nil, err
+					}
+					urls, err := uploadNLBWSCustomResources(&uploadCustomResourcesOpts{
```

Comment:
Just a callout here: we don't upload anything in `svc package` such as for addons and build and upload their image. However, I think we should because otherwise the parameter file we generate isn't complete. Maybe we can leave as is and i'll do some refactoring later again anyway.

Reply:
Side note:
From perusing the CDK help menu, it looks like `cdk synthesize` gives deterministic endpoints for their assets. For example, although the command doesn't upload things to S3 it prepopulates the params with the hash (the same parameter value is then generated with `deploy`). 

I think we can do that for our S3 artifacts as well, but not for ECR's image digest :( For a digest it looks like the image needs to be in a repository, maybe we can force them to provide a `--tag/--digest` flag in this scenario or `--upload-assets`

Reply:
That'd mean that until `package` is supported, people won't be able to use NLB through pipeline. Is that right?

Reply:
My proposal is to log an error message in `svc package` if it's run with an NLBWS that needs custom resources. In pipeline, it errors out and say "NLBWS using TLS or aliases isn't supported in pipeline yet". I will try to figure out a way to so that:
1. When people run `copilot svc package` manually, it doesn't error out and still produces a template; however, it spits out a friendly message that says the template doesn't work because the custom resource isn't uploaded.
2. In pipeline, it should fail at the build stage and says "NLBWS using TLS or aliases isn't supported in pipeline yet"

What do you think bosses?

Reply:
Chatted w @iamhopaul123 earlier, but one thing that we can do to unblock this PR:
1. Have NLB custom resources behave like other existing custom resources by minimizing and inlining the code in the CFN template
2. In order to avoid hitting the template body limit of 51KB (https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cloudformation-limits.html), we can create a separate PR that uploads the template to S3 instead (1MB limit):
   * Maybe under this path: `manual/templates/<service name>/<sha256 hash of template body>.yml`
   * We can merge the upload PR before this one in order to unblock this PR